system/Model.php

```diff
@@ -642,44 +645,55 @@ public function getInsertID(): int
 	 * @return integer|string|boolean
 	 * @throws \ReflectionException
 	 */
-	public function insert($data = null, bool $returnID = true)
+	public function insert(&$data = null, bool $returnID = true)
 	{
 		$escape = null;
 
 		$this->insertID = 0;
 
-		if (empty($data))
+		// Copy data array, so we don't lose
+		// object reference to $data var in case of updating exist state.
+		$rawData = $data;
+
+		if (empty($rawData))
 		{
-			$data           = $this->tempData['data'] ?? null;
+			$rawData        = $this->tempData['data'] ?? null;
 			$escape         = $this->tempData['escape'] ?? null;
 			$this->tempData = [];
 		}
 
-		if (empty($data))
+		if (empty($rawData))
 		{
 			throw DataException::forEmptyDataset('insert');
 		}
 
 		// If $data is using a custom class with public or protected
 		// properties representing the table elements, we need to grab
 		// them as an array.
-		if (is_object($data) && ! $data instanceof stdClass)
+		if (is_object($rawData) && ! $rawData instanceof stdClass)
 		{
-			$data = static::classToArray($data, $this->primaryKey, $this->dateFormat, false);
+			$tryUpdate = method_exists($rawData, 'exists') && $rawData->exists();
```

Comment:
Why do you feel lit's necessary to check for an update in the middle of the insert method? To me, that just hides errors that a developer needs to address in their app...

Reply:
yeah your complety right, i would just check the exists state in the save method not in the insert method, because only when you use the save method, you want to save an entity and you propaply dont know its state by your own. And when you use insert and update method, your pretty sure you want to update or insert the data set, in case of update by a given key.